test/yagna/module/payments/test_zero_amount_txs.py

```diff
@@ -0,0 +1,147 @@
+            volumes={assets_path / "requestor": "/asset"},
+            environment=requestor_env,
+            payment_id=payment_id_pool.get_id(),
+        ),
+        YagnaContainerConfig(
+            name="provider",
+            probe_type=ProviderProbeWithLogSteps,
+            environment=provider_env,
+            # https://github.com/golemfactory/goth/issues/410
+            privileged_mode=True,
+            volumes=provider_volumes,
+        ),
+    ]
+
+
+@pytest.mark.asyncio
+async def test_zero_amount_invoice_is_settled(
+    assets_path: Path,
+    demand_constraints: str,
+    exe_script: dict,
+    payment_id_pool: PaymentIdPool,
+    runner: Runner,
+    task_package_template: str,
+):
+    """Test successful flow requesting WASM tasks with goth REST API client."""
+
+    topology = _topology(assets_path, payment_id_pool)
+
+    async with runner(topology):
+        requestor = runner.get_probes(probe_type=RequestorProbeWithApiSteps)[0]
+        provider = runner.get_probes(probe_type=ProviderProbeWithLogSteps)[0]
+
+        # Market
+
+        await provider.wait_for_offer_subscribed()
+
+        task_package = task_package_template.format(
+            web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+        )
+
+        subscription_id, demand = await requestor.subscribe_demand(
+            task_package, demand_constraints
+        )
+
+        proposal = (
+            await requestor.wait_for_proposals(
+                subscription_id,
+                [provider],
+                lambda proposal: (
+                    proposal.properties.get("golem.runtime.name") == "wasmtime"
+                ),
+            )
+        )[0]
+        logger.info("Collected the proposal")
+
+        logger.info("Processing proposal from %s", provider.name)
+
+        # HINT: According to @kubam this is needed.
+        counterproposal_id = await requestor.counter_proposal(
+            subscription_id, demand, proposal
+        )
+        await provider.wait_for_proposal_accepted()
+
+        new_proposals = await requestor.wait_for_proposals(subscription_id, (provider,))
+        new_proposal = new_proposals[0]
+        assert new_proposal.prev_proposal_id == counterproposal_id
+
+        # Here: Agreement
+        agreement_id = await requestor.create_agreement(new_proposal)
+        await requestor.confirm_agreement(agreement_id)
+        await provider.wait_for_agreement_approved()
+
+        await requestor.unsubscribe_demand(subscription_id)
+        logger.info("Got agreement")
+
+        #  Zero-amount invoice is issued when agreement is terminated without activity
+
+        logger.info("Waiting for the agreement termination... ")
+        # When agreement termination is supported replace the sleep with below lines
+        await sleep(95)  # waiting for agreement timeout
+        # await requestor.terminate_agreement(agreement_id, None)
+        # await provider.wait_for_agreement_terminated()
```

Comment:
The `sleep` here is not acceptable. We don't want this much idle overhead on our CI test runs.

Instead, I'd suggest to just add the proper `terminate_agreement` implementation. You can add these two methods to their respective classes, they will be analogous to what's already there. For `requestor.terminate_agreement` take a look at:
https://github.com/golemfactory/goth/blob/adf441cbe07e1bbf8a9c44b10ba86ef103455064/goth/runner/requestor.py#L166-L169
Just create a new function and replace `confirm_agreement` with `terminate_agreement` in the client call.
For the provider log assertion:
https://github.com/golemfactory/goth/blob/adf441cbe07e1bbf8a9c44b10ba86ef103455064/goth/runner/provider.py#L37-L40
Add a new function like the one above and replace the log regex with the one you need.